src/components/PrometheusGraph/PrometheusGraph.js

```diff
@@ -351,18 +353,34 @@ class PrometheusGraph extends React.PureComponent {
       })),
     }));
 
-    this.setState({ multiSeries });
+    this.setState({ multiSeries }, () => {
+      // time and value scales use state.multiSeries so must be calculated after
+      // state.multiseries
+      this.prepareTimeAndValueScales(props);
+    });
```

Comment:
Isn't it possible that this part will only be executed in the next render cycle?